### PR TITLE
use logical comparisons instead of binary ones

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Replace binary comparisons of VPack Slices with logical ones.

- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behaviour change can only be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as all cluster tests.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5020/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue

File: arangod/Agency/Job.cpp [hunk 1]
```diff
@@ -490,9 +489,9 @@ std::string Job::findNonblockedCommonHealthyInSyncFollower(  // Which is in "GOO
       bool found = false;
       for (const auto& plannedServer :
            VPackArrayIterator(snap.hasAsArray(plannedShardPath).first)) {
-        if (plannedServer == server) {
+        if (basics::VelocyPackHelper::compare(plannedServer, server, false) == 0) {
           found = true;
-          continue;
+          break;
```
Comment (maierlars) [inline arangod/Agency/Job.cpp]: WOW!

File: arangod/Agency/Store.cpp [hunk 2]
```diff
@@ -469,9 +469,9 @@ check_ret_t Store::check(VPackSlice const& slice, CheckMode mode) const {
             if (node->slice().isArray()) {
               bool _found = false;
               for (auto const& i : VPackArrayIterator(node->slice())) {
-                if (i == op.value) {
+                if (basics::VelocyPackHelper::compare(i, op.value, false) == 0) {
                   _found = true;
-                  continue;
+                  break;
```
Comment (maierlars) [inline arangod/Agency/Store.cpp]: This is hilarious :D

File: arangod/Cluster/ClusterInfo.cpp [hunk 3]
```diff
@@ -2686,7 +2686,7 @@ void ClusterInfo::loadCurrentDBServers() {
         if (cleanedDBServers.isArray()) {
           bool found = false;
           for (auto const& cleanedServer : VPackArrayIterator(cleanedDBServers)) {
-            if (dbserver.key == cleanedServer) {
+            if (basics::VelocyPackHelper::compare(dbserver.key, cleanedServer, false) == 0) {
               found = true;
               continue;
```
Comment (maierlars) [inline arangod/Cluster/ClusterInfo.cpp]: I found another one! :+1:

File: arangod/Agency/Job.cpp (hunk 1 above)
Comment (fceller) [reply arangod/Agency/Job.cpp]: LGTM

File: arangod/Cluster/ClusterInfo.cpp (hunk 3 above)
Comment (jsteemann) [reply arangod/Cluster/ClusterInfo.cpp]: Fixed

**Comment by jsteemann** [comment]
Re-running tests after latest change:
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5050/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5063/

**Pull request merged**